src/main/java/com/store/book/service/user/impl/UserServiceImpl.java

```diff
@@ -39,4 +42,14 @@ public UserResponseDto register(UserRegistrationRequestDto request)
         User savedUser = userRepository.save(user);
         return userMapper.toUserResponse(savedUser);
     }
+
+    @Override
+    public Optional<User> getCurrentUser() {
+        Authentication authentication = SecurityContextHolder.getContext().getAuthentication();
```

Comment:
using Spring Context inside your business logic is a bad practice. 

it tight coupling your app and spring framework.
One of the problems of tight coupling is complex testing.